microservices/recordings/src/Service/Encoder.php

```diff
@@ -25,292 +24,372 @@ class Encoder
-                $file = $matches[0];
-                $callid = urldecode($matches[1]);
-            } else {
-                $this->logger->error(sprintf("[Recordings][ERROR] Unable to get info from filename %s\n", $filename));
-                continue;
+    private function processAccCdr(BillableCallInterface|UsersCdrInterface $accCdr, RawRecordingInfo $rawRecordingInfo): ?RecordingDto
+    {
+        $convertWav = $this->rawRecordingsDir . $rawRecordingInfo->getFileName();
+        $convertMp3 = $this->rawRecordingsDir . $rawRecordingInfo->getHashid() . ".mp3";
+        $metadata = 'artist="' . $rawRecordingInfo->getCallid() . '"';
+
+        if ($accCdr instanceof BillableCallInterface) {
+            $recordingDto = $this->getRecordingFromBillableCall($accCdr);
+            if (is_null($recordingDto)) {
+                return null;
+            }
+        } elseif ($accCdr instanceof UsersCdrInterface) {
+            $recordingDto = $this->getRecordingFromUserCdr($accCdr);
+            if (is_null($recordingDto)) {
+                return null;
             }
+        } else {
+            // This should not even be possible
+            $this->logger->error(sprintf(
+                "[Recordings][ERROR] Invalid CDR entries for %s\n",
+                $rawRecordingInfo->getCallid()
+            ));
+            return null;
+        }
 
-            // Get Call-Id hash id
-            $hashid = substr(md5($callid), 0, 8);
+        $this->logger->info(sprintf(
+            "[Recordings][%s] Encoding to %s\n",
+            $rawRecordingInfo->getHashid(),
+            basename($convertMp3)
+        ));
+
+        $duration = $this->convertToMp3(
+            $rawRecordingInfo,
+            $convertWav,
+            $metadata,
+            $convertMp3
+        );
 
-            // Get callid from file
-            $this->logger->info(sprintf("[Recordings][%s] Checking file %s\n", $hashid, $file));
+        if ($duration === 0.0) {
+            return null;
+        }
 
-            // Look if the conversation with that id has ended
-            $kamAccCdrs = [];
-            $kamAccCdrs = array_merge(
-                $kamAccCdrs,
-                $this->billableCallRepository->findByCallid($callid)
+        $recordingDto->setDuration(ceil($duration));
+        $recordingDto->setRecordedFilePath($convertMp3);
+
+        return $recordingDto;
+    }
+
+    private function convertToMp3(RawRecordingInfo $rawRecordingInfo, string $convertWav, string $metadata, string $convertMp3): float
+    {
+        try {
+            $mp3info = $this
+                ->rawRecordingProcessor
+                ->execute($convertWav, $metadata, $convertMp3);
+        } catch (\Exception $e) {
+            $this->stats['error']++;
+            $this->logger->info(sprintf(
+                "[Recordings][%s] Failed to convert audio: %s\n",
+                $rawRecordingInfo->getHashid(),
+                $e->getMessage()
+            ));
+
+            return 0;
```

Comment:
This function always returns 0, but is not the duration of the mp3 the expected return value?

Reply:
My bad, didn't check the diff properly